zulipterminal/ui_tools/buttons.py

```diff
@@ -344,25 +346,47 @@ def __init__(
             prefix_character="",
             show_function=self.update_emoji_button,
         )
-        if self._has_user_reacted_to_msg() or is_selected(self.emoji_name):
-            self.update_widget((None, f" {CHECK_MARK} "), None)
+
+        has_check_mark = self._has_user_reacted_to_msg() or is_selected(self.emoji_name)
+        self.update_widget((None, self.get_update_widget_text(has_check_mark)), None)
 
     def _has_user_reacted_to_msg(self) -> bool:
         return self.controller.model.has_user_reacted_to_message(
             self.message, emoji_code=self.emoji_code
         )
 
+    def get_update_widget_text(self, user_reacted: bool) -> str:
+        count_text = str(self.reaction_count) if self.reaction_count > 0 else ""
+        reacted_check_mark = CHECK_MARK if user_reacted else ""
+        return f" {reacted_check_mark} {count_text} "
+
     def update_emoji_button(self) -> None:
         if self._has_user_reacted_to_msg():
-            if self.is_selected(self.emoji_name):
-                self.update_widget((None, f" {CHECK_MARK} "), None)
-            else:
-                self.update_widget((None, ""), None)
+            self.reaction_count = (
+                (self.reaction_count + 1)
+                if self.is_selected(self.emoji_name)
+                else (self.reaction_count - 1)
+            )
+            self.update_widget(
+                (
+                    None,
+                    self.get_update_widget_text(self.is_selected(self.emoji_name)),
+                ),
+                None,
+            )
         else:
-            if self.is_selected(self.emoji_name):
-                self.update_widget((None, ""), None)
-            else:
-                self.update_widget((None, f" {CHECK_MARK} "), None)
+            self.reaction_count = (
+                (self.reaction_count - 1)
+                if self.is_selected(self.emoji_name)
+                else (self.reaction_count + 1)
```

Comment:
If you invert this then this becomes identical to the other main branch except for the `[not] self.is_selected(...)`.

Reply:
You've amended this slightly, but note how if you extract the piece I highlighted in the comment as a variable, then you can collapse the branches into one.